### ISSUE TITLE
It can't browse this repo's bookmark

### ISSUE THREAD

**Issue by zheneknew** [xwmx/nb#179]
Commands to reproduce:
```
nb notebooks new test
nb use test
nb bk https://github.com/xwmx/nb
BROWSER=qutebrowser nb browse --gui 1
nb b 1
```
It stucks at 20% loading and starts to roast CPU, with any browser. 
Screencast: https://vimeo.com/718965571

I've tried to do this with some other long pages (e.g. https://core.telegram.org/bots/api or https://datatracker.ietf.org/doc/html/rfc2616) and it works fine.
So there is some element on this repo page that the nb underhood server is not able to serve.

```
nb --version
6.11.1
```

**Comment by xwmx** [comment]
@zheneknew Thanks for the heads up. It's currently working on my macOS system. What is your `pandoc --version`? Mine is 2.18, which appears to be the latest.

**Comment by zheneknew** [comment]
```
pandoc --version
pandoc 2.14.0.3
Compiled with pandoc-types 1.22, texmath 0.12.3, skylighting 0.10.5.2,
citeproc 0.4.0.1, ipynb 0.1.0.1
```
This version is latest in dnf - fedora's official repos

**Comment by xwmx** [comment]
I was able to reproduce the issue in Fedora 36 with pandoc 2.14.0.3.

The following command prints the trace (where `1` represents the id of the bookmark):

```bash
cat "$(nb show 1 --path)" | pandoc -f markdown -t html --verbose --trace
```

According to this, these problematic line is `128` which consists of a newline followed by an empty markdown list item on line `129`:

```md

-  
```

Removing either the newline or the empty list item from the document allows it to process like normal. I don't currently know why this error is occurring.

Using pandoc version 2.18 ([from here](https://github.com/jgm/pandoc/releases/tag/2.18)) does not resolve the issue on Fedora. Pandoc 2.18 on macOS appears to work fine.

**Comment by zheneknew** [comment]
I've tried to do this in archlinux with pandoc installed from pacman and with last pandoc 2.18, and it fails too. So I think it's common problem for all linux distros.
It may make sense to compare files generated from macos and from linux.
Could you make `diff file1 file2` with file generated in macos?(i haven't access  to macs)
here is file just generated in fedora 35 with pandoc-2.14
[20220611083536.bookmark.md](https://github.com/xwmx/nb/files/8883329/20220611083536.bookmark.md)

If there is a difference, it also makes sense to try to serve a file generated in linux in macos. 
All this will mean that the problem appears at the time of the bookmark generation

**Comment by xwmx** [comment]
It was an issue on macOS, too, just masked in my environment by Readability. It looks like the empty list item might be the issue. I added a `sed` filter to strip out empty list items when processing bookmark content. Bookmarking the page on Fedora now creates a bookmark that successfully renders in `browse`. Note that this doesn't resolve the issue with existing bookmarks, yet. I'm still considering whether to add a filter for this on the rendering side.

**Comment by xwmx** [comment]
The bookmarking side of the issue should now be resolved as of version 6.11.2. This seems like enough of an edge case that I’m going to skip adding a filter on the rendering side for now. In my testing, the length of the content after the empty list item is part of what’s causing it, and it doesn’t seem to be an issue with shorter documents containing empty list items.


Please let me know if you run into any more issues with it or other thoughts. Thanks!
